External/FEXCore/Source/Interface/Core/CPUID.cpp

```diff
@@ -77,18 +77,30 @@ static uint32_t GetCPUID() {
 static uint32_t CalculateNumberOfCPUs() {
   size_t CPUs = 1;
 
-  while(std::filesystem::exists("/sys/devices/system/cpu/cpu" + std::to_string(CPUs))) {
-    CPUs++;
+#define CPU_PATH "/sys/devices/system/cpu/cpu"
+
+  constexpr size_t CPU_PATH_SIZE = sizeof(CPU_PATH) - 1;
+  char MaxPath[] = CPU_PATH "\0\0\0\0";
+  // Gives us a maximum of 999 CPU cores
+  constexpr size_t MAX_SIZE = sizeof(MaxPath);
+  static_assert(MAX_SIZE == 32, "Fixed max");
+#undef CPU_PATH
+
+  for (CPUs = 1; CPUs <= 999; ++CPUs) {
+    fmt::format_to_n(&MaxPath[CPU_PATH_SIZE], 3, "{}", CPUs);
+    if (access(MaxPath, F_OK) != 0) {
+      return CPUs - 1;
+    }
```

Comment:
Instead of using this much more verbose logic, wouldn't this be a good place to use a static local variable?

```suggestion
  static size_t CPUs = std::invoke([]() {
      size_t CPUs = 1;
      while(std::filesystem::exists("/sys/devices/system/cpu/cpu" + std::to_string(CPUs))) {
        CPUs++;
      }
      return CPUs;
    });
```

Reply:
Just dropped this change fragment for now.